--- live-build-20190311.orig/scripts/build/binary_rootfs	2019-03-11 11:05:11.000000000 +0100
+++ live-build-20190311/scripts/build/binary_rootfs	2019-08-12 08:15:55.926804968 +0200
@@ -318,8 +318,6 @@
 		# Remove stale squashfs image
 		rm -f chroot/filesystem.squashfs
 
-		MKSQUASHFS_OPTIONS="${MKSQUASHFS_OPTIONS} -no-progress"
-
 		if [ "${_VERBOSE}" = "true" ]
 		then
 			MKSQUASHFS_OPTIONS="${MKSQUASHFS_OPTIONS} -info"
@@ -340,7 +338,7 @@
 			esac
 		fi
 
-		MKSQUASHFS_OPTIONS="${MKSQUASHFS_OPTIONS} -comp xz"
+		MKSQUASHFS_OPTIONS="${MKSQUASHFS_OPTIONS} -comp xz -Xbcj x86"
 
 		case "${LB_BUILD_WITH_CHROOT}" in
 			true)
