Index: live-build-20190311/scripts/build/chroot_hooks
===================================================================
--- live-build-20190311.orig/scripts/build/chroot_hooks
+++ live-build-20190311/scripts/build/chroot_hooks
@@ -78,12 +78,14 @@ then
 fi
 
 ## Processing local hooks
-if ls config/hooks/normal/*.chroot config/hooks/live/*.chroot > /dev/null 2>&1
+if ls config/hooks/normal/*.chroot > /dev/null 2>&1 ||
+   ls config/hooks/live/*.chroot > /dev/null 2>&1 ||
+   ls config/hooks/live/*.container > /dev/null 2>&1
 then
 	# Restoring cache
 	Restore_cache cache/packages.chroot
 
-	for HOOK in config/hooks/normal/*.chroot config/hooks/live/*.chroot
+	for HOOK in config/hooks/normal/*.chroot config/hooks/live/*.chroot config/hooks/live/*.container
 	do
 		if [ ! -e "${HOOK}" ]
 		then
@@ -100,7 +102,17 @@ then
 		fi
 
 		# Executing hook
-		Chroot chroot "/root/$(basename ${HOOK})" || { Echo_error "${HOOK} failed (exit non-zero). You should check for errors."; exit 1 ;}
+		Echo_message "Executing hook ${HOOK}..."
+		case ${HOOK} in
+			*.container)
+				umount chroot/proc
+				systemd-nspawn --capability=all -D chroot "/root/$(basename ${HOOK})" || { Echo_error "${HOOK} failed (exit non-zero). You should check for errors."; exit 1 ;}
+				mount proc-live -t proc chroot/proc
+				;;
+			*)
+				Chroot chroot "/root/$(basename ${HOOK})" || { Echo_error "${HOOK} failed (exit non-zero). You should check for errors."; exit 1 ;}
+				;;
+		esac
 
 		# Removing hook
 		rm -f chroot/root/"$(basename ${HOOK})"
