diff --git a/scripts/build/chroot b/scripts/build/chroot
index 497911800..57801e3c8 100755
--- a/scripts/build/chroot
+++ b/scripts/build/chroot
@@ -37,6 +37,7 @@ lb chroot_prep install all mode-archives-chroot "${@}"
 lb chroot_linux-image "${@}"
 lb chroot_firmware "${@}"
 lb chroot_preseed "${@}"
+lb chroot_includes_before_packages ${@}
 
 for _PASS in install live; do
 	lb chroot_package-lists ${_PASS} "${@}"
@@ -47,7 +48,7 @@ for _PASS in install live; do
 	fi
 done
 
-lb chroot_includes "${@}"
+lb chroot_includes_after_packages ${@}
 lb chroot_hooks "${@}"
 lb chroot_hacks "${@}"
 lb chroot_interactive "${@}"
diff --git a/scripts/build/chroot_includes_after_packages b/scripts/build/chroot_includes_after_packages
new file mode 100755
index 000000000..f781df1de
--- /dev/null
+++ b/scripts/build/chroot_includes_after_packages
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+## live-build(7) - System Build Scripts
+## Copyright (C) 2016-2020 The Debian Live team
+## Copyright (C) 2006-2015 Daniel Baumann <mail@daniel-baumann.ch>
+##
+## This program comes with ABSOLUTELY NO WARRANTY; for details see COPYING.
+## This is free software, and you are welcome to redistribute it
+## under certain conditions; see COPYING for details.
+
+
+set -e
+
+# Including common functions
+[ -e "${LIVE_BUILD}/scripts/build.sh" ] && . "${LIVE_BUILD}/scripts/build.sh" || . /usr/lib/live/build.sh
+
+# Setting static variables
+DESCRIPTION="Copy files into chroot after packages have been installed"
+USAGE="${PROGRAM} [--force]"
+
+# Processing arguments and configuration files
+Init_config_data "${@}"
+
+Echo_message "Begin copying chroot includes after packages have been installed..."
+
+# Requiring stage file
+Require_stagefiles config bootstrap
+
+# Checking stage file
+Check_stagefile
+
+# Acquire lock file
+Acquire_lockfile
+
+if Find_files config/includes.chroot_after_packages/
+then
+	# Copying includes
+	cd config/includes.chroot_after_packages
+	Echo_message "Creating a tarball with files from includes.chroot_after_packages..."
+	tar cf "${OLDPWD}"/chroot/includes.chroot.tar .
+	cd "${OLDPWD}"
+	Echo_message "Extracting the tarball in the chroot..."
+	Chroot chroot "tar -xvf includes.chroot.tar --no-same-owner --keep-directory-symlink --overwrite"
+	rm chroot/includes.chroot.tar
+fi
+
+# Creating stage file
+Create_stagefile
diff --git a/scripts/build/chroot_includes b/scripts/build/chroot_includes_before_packages
similarity index 82%
rename from scripts/build/chroot_includes
rename to scripts/build/chroot_includes_before_packages
index dc4bb6993..5601057aa 100755
--- a/scripts/build/chroot_includes
+++ b/scripts/build/chroot_includes_before_packages
@@ -15,13 +15,13 @@ set -e
 [ -e "${LIVE_BUILD}/scripts/build.sh" ] && . "${LIVE_BUILD}/scripts/build.sh" || . /usr/lib/live/build.sh
 
 # Setting static variables
-DESCRIPTION="Copy files into chroot"
+DESCRIPTION="Copy files into chroot before packages have been installed"
 USAGE="${PROGRAM} [--force]"
 
 # Processing arguments and configuration files
 Init_config_data "${@}"
 
-Echo_message "Begin copying chroot includes..."
+Echo_message "Begin copying chroot includes before packages have been installed..."
 
 # Requiring stage file
 Require_stagefiles config bootstrap
@@ -32,10 +32,10 @@ Check_stagefile
 # Acquire lock file
 Acquire_lockfile
 
-if Find_files config/includes.chroot/
+if Find_files config/includes.chroot_before_packages/
 then
 	# Copying includes
-	cd config/includes.chroot
+	cd config/includes.chroot_before_packages
 	Echo_message "Creating a tarball with files from includes.chroot..."
 	tar cf "${OLDPWD}"/chroot/includes.chroot.tar .
 	cd "${OLDPWD}"
