From 4276c1b887cfb11e141874015b8b7b6aacf6dcaf Mon Sep 17 00:00:00 2001
From: Ronny Standtke <ronny.standtke@gmx.net>
Date: Tue, 18 Aug 2026 17:42:24 +0200
Subject: [PATCH] try to open LUKS devices with a default password

---
 components/9990-misc-helpers.sh | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/components/9990-misc-helpers.sh b/components/9990-misc-helpers.sh
index 1920ecc..bb57c8b 100755
--- a/components/9990-misc-helpers.sh
+++ b/components/9990-misc-helpers.sh
@@ -906,6 +906,14 @@ open_luks_device ()
 		fi
 	fi
 
+	# Try to open the device non-interactively with the default passphrase "live"
+	if echo -n "live" | cryptsetup -T 1 luksOpen ${dev} ${name} ${opts}
+	then
+		luks_device="/dev/mapper/${name}"
+		echo ${luks_device}
+		return 0
+	fi
+
 	load_keymap
 
 	# check for plymouth
-- 
2.47.3

