diff --git a/components/9990-misc-helpers.sh b/components/9990-misc-helpers.sh
index 595c7f0..05a50d1 100755
--- a/components/9990-misc-helpers.sh
+++ b/components/9990-misc-helpers.sh
@@ -1101,9 +1101,9 @@ find_persistence_media ()
 	# in one union together.
 	#
 	black_listed_devices=""
-	for d in /run/live/rootfs/* /run/live/findiso /run/live/fromiso
+	for d in /run/live/medium /run/live/rootfs/* /run/live/findiso /run/live/fromiso
 	do
-		black_listed_devices="${black_listed_devices} $(what_is_mounted_on d)"
+		black_listed_devices="${black_listed_devices} $(what_is_mounted_on $d)"
 	done
 
 	for dev in $(storage_devices "${black_listed_devices}" "${white_listed_devices}")
